Update MAAS icon

sreejaPaul opened the pull request:
**Preview:**
<img width="975" height="664" alt="image" src="https://github.com/user-attachments/assets/7877fc35-09f5-4178-8733-80f397fb804f" />


**Issue:** closes #13953

**Popularity metric:**

**Terms of Service link:**
https://canonical.com/legal/terms-and-policies

### Checklist

- [x] I have reviewed the [forbidden brands](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#forbidden-brands) list and confirm the brand I am submitting a PR for is not one of them, nor is it a subsidiary of one of those brands
- [x] I have reviewed the brand's terms of service, and am confident we can add this icon
- [x] I updated the JSON data in `data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### Description

1. Since this is an icon update, I have kept the name of the icon in the `data/simple-icons.json` file as it was previously (MAAS) rather than changing it to Canonical MAAS.
2. Icon color `#E95420` is taken from the core color pallet of the [Canonical/Ubuntu brand palette](https://design.ubuntu.com/brand/colour-palette).
3. Icon is sourced from [official design asset](https://assets.ubuntu.com/v1/e45b7f3f-Canonical%20MAAS.svg) and is listed [here](https://design.ubuntu.com/resources) in the logos section.
4. Followed [Simple Icon's official contribution documentation](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#adding-or-updating-an-icon) and [extracted](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#2-extract-the-icon-from-the-logo) the icon using InkSpace and [optimized](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#3-optimize-the-icon) using SVGOMG Online Tool, along with other guidelines.
5. All tests and lint checks passed locally after adding this icon.

vosdev commented:
MAAS icon already exists so it would be an update instead of an addition. I made an accident in the initial feature request. Updated the title but was unable to update the label to the issue.

sreejaPaul commented:
Hi @cscnk52 @adamrusted, 

1. As this is now an icon update issue, I have adjusted the code accordingly — removed previously added code and added the respective icon update code.
2. Updated the icon preview image, the PR description in [this comment](https://github.com/simple-icons/simple-icons/pull/14005#issue-3637977439).
3. Given these points [in my earlier comment](https://github.com/simple-icons/simple-icons/pull/14005#issuecomment-3554789916) I’d appreciate guidance from the team on which metrics, if any, should be considered for this icon. That’s why I haven’t put anything here yet  in [this comment](https://github.com/simple-icons/simple-icons/pull/14005#issue-3637977439).
4. Per earlier feedback from @adamrusted , I modified the icon design to include the rectangle part, and this change has been included in the PR.

Note: As the issue changed from adding to updating after the PR was raised, one commit and the branch name still include “add,” while later commits and other references use “update.” This is kept for continuity; the PR content reflects the icon update.
